### DISCUSSION TITLE
How to push a new ObjectId to an array?

### DISCUSSION THREAD

**Discussion by AnthonyLzq** [prisma/prisma#10876] [Q&A] [answered]
I have the following schema

```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["mongoDb"]
}

datasource db {
  provider = "mongodb"
  url      = env("MONGO_URI")
}

model Course {
  id          String   @id @default(dbgenerated()) @map("_id") @db.ObjectId
  description String
  teacher     String
  title       String
  topic       String
  students    String[] @db.Array(ObjectId)
}

model Student {
  id    String @id @default(dbgenerated()) @map("_id") @db.ObjectId
  name  String
  email String
}
```

Want I want to achieve is to enroll new students to a course, so I want to push a new ObjectId to my students array, but Prisma fails with the following error:

```
The current database provider doesn't support a feature that the query used: Converting BSON to type String. Data: Array([ObjectId("61c8a8d2e2d9caa2a4f91638")])
```

This is the code I have:
```js
const enrollStudent = (courseId: string, studentId: string) => {
  return await prismaClient.course.update({
    where: {
      id: courseId
    },
    data: {
      students: {
        push: studentId
      }
    }
  })
}
```

According to the [docs](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#push) the push operation is only available for PostgreSQL, is there any plan to make it available for mongodb as well?

**Comment by dpetrick** [comment, marked as answer]
This is a bug. Tracking in: https://github.com/prisma/prisma/issues/10906

**Reply by dpetrick** [reply]
The fix is ready, it will be included in the upcoming 3.8 release.

**Reply by AnthonyLzq** [reply]
Hi there, the problem wasn't with `set`, instead with `push`, I have just updated the question :(

**Reply by pantharshit00** [reply]
I checked the PR and looks like `push` is also handled correctly now. Can you please try the dev release channel once? 

`npm install prisma@dev @prisma/client@dev`

**Reply by dpetrick** [reply]
Yes, the fix fixes push, not set.

**Reply by AnthonyLzq** [reply]
I can confirm it works, thank you!
